### ISSUE TITLE
Remove prototype elements when data is present

### ISSUE THREAD

**Issue by iamvery** [iamvery/ratchet#55]
Would like to be able to designate certain elements as prototype. This would allow multiple versions of a thing to be shown in prototype, but exclude it in the final render.

For example,

``` html
<div data-prop="foo">First</div>
<div data-proto="foo">Second</div>
```

When rendered with no data, the prototype would be included. However when rendered with data, it would look like:

``` elixir
data = %{foo: "lolwat"}
```

``` html
<div data-prop="foo">lolwat</div>
<!-- note the prototype element is excluded -->
```

**Comment by bryanp** [comment]
Interesting! I may steal this for Pakyow 😁

We have something called view versions that this would fit right into. For example:

``` html
<div data-prop="foo" data-version="default" class="foo">First</div>
<div data-prop="foo" data-version="alt" class="fooalt">First</div>
<div data-version="proto" class="fooproto">First</div>
```

When running in prototype mode, all versions are presented. Outside of that, `default` would be always be presented unless the backend chooses `alt`; `proto` would always be removed.

**Comment by dgmcguire** [comment]
I think there would be less surprise for users of ratchet if you just had multiple props of the same name and have merged data overwrite everything, rather than have a more explicit tool for specifying prototype data I would rather have everything that is inside a data-prop to be by default considered prototype that can be overwritten by merged in data.

**Comment by iamvery** [comment]
Seems like a good idea @dgmcguire. Glad you like the idea @bryanp 😊

**Comment by bryanp** [comment]
Hmm maybe I misunderstood the purpose of this feature. In Pakyow, the `innerContent` of the node is overridden when the data is presented. What I thought you were getting at here is that sometimes nodes are needed for decoration that would never be presenting data, and thus not be a `data-prop`. Can you back up and explain more of your thinking here?

**Comment by iamvery** [comment]
The use case I'm trying to address is sometimes a prototype view needs multiple example elements to really drive the point of it home. I suppose the canonical example might be an actual list. If the `<li>` had a data-prop, the designer would probably want to illustrate that portion of the view with multiple list items for example. However, for rendering, only the first or "non prototype" item would be used. Does that make sense?

**Comment by bryanp** [comment]
Yep, got it. In Pakyow, this is a non-issue. When you look for a prop by name, it returns an object that represents every matching view prop at that level. Then, the data presentation method handles removing them (essentially making the view look like the data). I believe this is what @dgmcguire is getting at as well. So, looks like I misunderstood 😬
